src/System.Net.WebSockets.Client/tests/ClientWebSocketOptionsTests.cs

```diff
@@ -67,6 +67,43 @@ public static void Proxy_Roundtrips()
             }
         }
 
+        [ActiveIssue(28027)]
+        [OuterLoop] // TODO: Issue #11345
+        [ConditionalTheory(nameof(WebSocketsSupported)), MemberData(nameof(EchoServers))]
+        public async Task Proxy_ConnectThruProxy_Success(Uri server)
+        {
+            string proxyServerUri = System.Net.Test.Common.Configuration.WebSockets.ProxyServerUri;
+            if (string.IsNullOrEmpty(proxyServerUri))
+            {
+                _output.WriteLine("Skipping test...no proxy server defined.");
```

Comment:
Can we use a local/loopback proxy instead of requiring one to be defined?

Reply:
The current loopback test proxy server here:
https://github.com/dotnet/corefx/blob/master/src/System.Net.Http/tests/FunctionalTests/LoopbackGetRequestHttpProxy.cs

doesn't support CONNECT tunneling. Nor does it support multiple requests, etc

I also wanted to be able to easily switch to any proxy server for these tests and this pattern allows that.

Reply:
What proxy are you using?  How does someone else run these tests?

Reply:
You could use Fiddler.  Or I have been using our 'HappySockets' server.  It is a simple standalone EXE that has various HTTP server and/or proxy server capabilities.  Tomas has also deployed a squid server.  We also have a corporate network only proxy server as well that we use for .NET Framework weekly test passes.

Reply:
>How does someone else run these tests?

And to run this test, just set the environment variable to the URI of the proxy server.  I.e.

```
set COREFX_WEBSOCKETPROXYSERVERURI=http://localhost:8080/
```

Reply:
Ok. FWIW, I'm fine if we want to write this in a way that allows substituting another proxy, but I still think the test should work without one configured.  If that means we need to add CONNECT tunneling support to a loopback proxy server we can run in process, fine.  These tests that require additional configuration effectively never get run after they're initially checked in.

Reply:
As a pattern to use for writing tests, it would be nice if we had a hybrid way of using a loopback server as well as picking a server from the environment variable.  That would give us the best of both worlds.

In terms of this: "These tests that require additional configuration effectively never get run after they're initially checked in"....that is already true for some of the more complex multi-machine tests with domain-joined machines etc. that are already checked into the repo.

Reply:
> that is already true for some of the more complex multi-machine tests with domain-joined machines etc. that are already checked into the repo

I know.  I've never run them once, and I expect that's true of pretty much everyone else on the team.  They've never run in CI.  They're effectively useless.

Reply:
That's why we need to figure out how to invest in a way that these kind of tests can be run more often by both automation (CI or nightly) and by a dev.